concurrent_test.go

```diff
@@ -0,0 +1,98 @@
+package bbolt_test
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	bolt "go.etcd.io/bbolt"
+	"go.etcd.io/bbolt/internal/btesting"
+)
+
+func TestRepeatableRead(t *testing.T) {
```

Comment:
@serathius Had a great comment that if we want to invest in API surface tests, we should consider the same linearizability library we use in etcd -> i.e. verify that there is linear order between reads and the interleaved writes.

Reply:
Good point. Let me do this in followup PR.